src/Makaira/Connect/Repository.php

```diff
@@ -323,13 +324,22 @@ protected function getRepositoryForType($type)
      *
      * @param string $type
      * @param string $id
+     *
+     * @throws Throwable
      */
     public function touch($type, $id)
     {
         if (!$id) {
             return;
         }
-        $this->database->execute($this->touchQuery, ['type' => $type, 'id' => $id]);
+
+        try {
+            $this->database->beginTransaction();
+            $this->database->execute($this->touchQuery, ['type' => $type, 'id' => $id]);
```

Comment:
A commit is missing here or did you do this on purpose to let the transaction automatically committed itself?
```suggestion
            $this->database->execute($this->touchQuery, ['type' => $type, 'id' => $id]);
            $this->database->commit();
```